main.go

```diff
@@ -316,22 +308,30 @@ func main() {
 }
 
 func registerControllers(ctx context.Context, mgr manager.Manager) {
+	machineCache, err := coalescing.NewRequestCache(5 * time.Second)
```

Comment:
Did you consider making this a configurable as a cmdline arg?

Reply:
I didn't, that would be a good one to be able to make configurable. Do you have any thoughts on whether it should be configurable per controller or just a single value?

Reply:
I had made it configurable for both AMP and AMPM. If it's not overkill for cmdline args, I'd vote for per controller.

Reply:
AMP and AMPM are both hardcoded right now, I don't think they're configurable.

Per controller makes sense, but it might be overwhelming to the user to be able to configure all of them (or to have to configure each one separately to change all the values). What do you think about one common flag for now, and potentially make it more granular later if the use case arises?

Reply:
You are right. I lied. I think I was thinking about doing that, but must have forgot or thought that I already had.

Reply:
wishful thinking :) 

okay let me know what you think of this:

1) using a common var across all the controllers: this is slightly less granular but I think it's a better, easier to understand configuration from a user's perspective (since they're not supposed to know how the code of each controller works), 10 seconds being a middle ground default value. 
2) the flag name "debouncing-timer" and description, tried to make that as human developer-friendly as possible and tried to describe how it's actually useful (ie. what it does) and not _how_ it does it (ie. "cache").